src/btree/bt_split.c

```diff
@@ -69,9 +69,10 @@ __split_verify_intl_key_order(WT_SESSION_IMPL *session, WT_PAGE *page)
     case WT_PAGE_COL_INT:
         recno = 0; /* Less than any valid record number. */
         WT_INTL_FOREACH_BEGIN (session, page, ref) {
-            WT_ASSERT(session, ref->home == page);
-
-            WT_ASSERT(session, ref->ref_recno > recno);
+            WT_ASSERT_ALWAYS(
+              session, ref->home == page, "Failed to verify page reference after splitting");
```

Comment:
I think we can have more detail here: "Internal page in illegal state, child ref is referencing an incorrect page"